### PR TITLE
Python support update

### LINKED ISSUE #5: Update supported Python versions
- drop support for Python versions < 3.6
- add support for Python versions 3.7, 3.8 (if missing)
- fix linting errors (if present)

If you are a beginner, and need to help for this issue, especially during Hacktoberfest, please reach out to me via https://jugmac00.github.io/

### PR DESCRIPTION
**Pull request by sgaist**
This merge request updates the minimal Python version to 3.6 and adds 3.8.

It also adds the tools required to run the tests to list of dependencies as well as fixes Python 3.7 warnings that are errors in Python 3.8.

Fixes #5

### CODE REVIEW

**Comment by jugmac00** [comment]
Hi @sgaist - thank you for your contribution!

Currently, CI fails - see https://travis-ci.org/github/morepath/morepath_wiki/builds/731753602

I will add some comments inline on how to fix the pr.

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
 
-python: 3.5
+python: 3.7
 env:
-    - TOXENV=py27
-    - TOXENV=py34
-    - TOXENV=py35
+    - TOXENV=py37
+    - TOXENV=py38
     - TOXENV=pep8
```
Comment (jugmac00) [inline .travis.yml:7]: As stated in #5 - I'd like to drop support for all versions below Python3.6 - so this means I do want to support Python 3.6. Maybe I should have made that clearer in the issue description.

Can you please add Python 3.6? Thank you.

Also, CI currently fails.

Travis has changed which interpreters are pre-installed on a machine. In order to test different Python versions, you have to use the `matrix` keyword. I was not aware of this until yesterday, when we faced the same problem in another repository.

Please update the above code similar to 
https://github.com/morepath/morepath_cerebral_todomvc/commit/c5b413728e05e92423c162534c32e6b39ee1cb0f

Thank you!

File: setup.py [hunk 2]
```diff
@@ -30,6 +30,8 @@
             'pytest',
             'pytest-cov',
             'webtest',
+            'tox',
+            'flake8'
         ],
```
Comment (jugmac00) [inline setup.py:35]: adding `flake8` here does not seem to be necessary, as it gets installed by tox, see this configuration:
```
[testenv:pep8]
basepython = python3
deps = flake8
commands = flake8 morepath_wiki setup.py
```

`deps` means install... a package

Adding `tox` here is probably ok. I have installed it globally, as every project I maintain uses tox, but if that helps you - go for it.

File: setup.py [hunk 3]
```diff
@@ -42,10 +44,7 @@
...
         'License :: OSI Approved :: BSD License',
         'Topic :: Internet :: WWW/HTTP :: WSGI',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.4',
-        'Programming Language :: Python :: 3.5',
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
     ]
```
Comment (jugmac00) [inline setup.py:49]: Please add Python 3.6

Thank you!

File: tox.ini [hunk 4]
```diff
@@ -1,17 +1,17 @@
 [tox]
-envlist = py27,py34,py35,pep8,coverage
+envlist = py37,py38,pep8,coverage
```
Comment (jugmac00) [inline tox.ini:3]: Please add Python 3.6, thank you!

**Review by jugmac00 — CHANGES_REQUESTED** [review]
Thanks for your PR - I have a couple of comments - please have a look at them and try to resolve them.

**Comment by jugmac00** [comment]
Could you also please add an entry to the change log? Thank you!

**Comment by jugmac00** [comment]
@sgaist CI is currently failing - the `matrix` thing, once again :-) see morepath/morepath_cerebral_todomvc@c5b4137

File: setup.py (hunk 2 above)
Comment (sgaist) [reply setup.py:34]: Done

File: tox.ini (hunk 4 above)
Comment (sgaist) [reply tox.ini:3]: Done

File: setup.py (hunk 3 above)
Comment (sgaist) [reply setup.py:49]: Done

File: .travis.yml (hunk 1 above)
Comment (sgaist) [reply .travis.yml]: Done

**Comment by jugmac00** [comment]
LGTM! Thanks for the contribution!

**Pull request merged**